// @azure/cosmos/dist-esm/src/encryption/EncryptionManager.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { ProtectedDataEncryptionKeyCache } from "./Cache/ProtectedDataEncryptionKeyCache";
import { KeyEncryptionKeyCache } from "./Cache/KeyEncryptionKeyCache";
import { EncryptionSettingsCache } from "./Cache/EncryptionSettingsCache";
import { ClientEncryptionKeyPropertiesCache } from "./Cache/ClientEncryptionKeyPropertiesCache";
import { EncryptionKeyStoreProvider } from "./EncryptionKeyStoreProvider";
import { Constants } from "../common/constants";
/**
 * Cache manager for encryption related caches.
 * @hidden
 */
export class EncryptionManager {
    constructor(encryptionKeyResolver, cacheTimeToLive) {
        this.cacheTimeToLive =
            cacheTimeToLive !== undefined
                ? cacheTimeToLive
                : Constants.DefaultEncryptionCacheTimeToLiveInSeconds;
        const cacheTtlInMs = this.getCacheTTlInMs();
        this.encryptionKeyStoreProvider = new EncryptionKeyStoreProvider(encryptionKeyResolver, cacheTtlInMs);
        this.protectedDataEncryptionKeyCache = new ProtectedDataEncryptionKeyCache(cacheTtlInMs);
        this.keyEncryptionKeyCache = new KeyEncryptionKeyCache();
        this.encryptionSettingsCache = new EncryptionSettingsCache();
        this.clientEncryptionKeyPropertiesCache = new ClientEncryptionKeyPropertiesCache();
    }
    /**
     * Converts the EncryptionTimeToLive instance to a number (milliseconds).
     */
    getCacheTTlInMs() {
        const millisecondsPerSecond = 1000;
        return Number(this.cacheTimeToLive * millisecondsPerSecond);
    }
}
